pkg/cmd/util/util.go

```diff
@@ -78,29 +96,90 @@ func (c *GRPCServerConfig) Complete(level zerolog.Level, svcRegistrationFn func(
 		listenFunc: func() error {
 			return srv.Serve(l)
 		},
+		dial: dial,
 		prestopFunc: func() {
 			log.WithLevel(level).Str("addr", c.Address).Str("network", c.Network).
 				Str("prefix", c.flagPrefix).Msg("grpc server stopped listening")
 		},
 		stopFunc: srv.GracefulStop,
-		enabled:  c.Enabled,
+		creds:    clientCreds,
 	}, nil
 }
 
+func (c *GRPCServerConfig) listenerAndDialer() (l net.Listener, dial DialFunc, err error) {
+	if c.Network == "bufconn" {
+		bl := bufconn.Listen(c.BufferSize)
+		dial = func(ctx context.Context, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+			opts = append([]grpc.DialOption{grpc.WithContextDialer(func(ctx context.Context, s string) (net.Conn, error) {
+				return bl.DialContext(ctx)
+			})}, opts...)
+			return grpc.DialContext(ctx, "bufnet", opts...)
+		}
+		l = bl
+		return
```

Comment:
I was a little confused that this mixed and match returning named params with explicit returns

Reply:
The functions that are defined inside the `listenerAndDialer` use explicit returns, the method `listenerAndDialer` itself uses named params. Do you think they should match?

Reply:
I think everything should be explicit returns because it's harder to read. I can treat the `if c.Network == BufferedNetwork` conditional as standalone scope-wise if not for the named return arguments.

Reply:
updated